src/test/groovy/graphql/nadel/NadelTest.groovy

```diff
@@ -107,4 +115,54 @@ class NadelTest extends Specification {
         }
     }
 
+
+    def "register special DataFetcher"() {
+        def dsl = """
+        service FooService {
+            type Query {
+                foo: Foo
+            }
+
+            type Foo {
+                specialField: String
+            }
+        }
+        """
+
+        def graphqlRemoteRetriever1 = Mock(GraphQLRemoteRetriever)
+        def callerFactory = mockCallerFactory([FooService: graphqlRemoteRetriever1])
+
+        TypeDefinitionRegistry typeDefinitionRegistry = new TypeDefinitionRegistry();
+
+
+        def fieldDefinition = FieldDefinition
+                .newFieldDefinition()
+                .name("specialField")
+                .type(new TypeName("String"))
+                .build()
+        ObjectTypeDefinition objectTypeDefinition = ObjectTypeDefinition.newObjectTypeDefinition()
```

Comment:
I am thinking that we can do better: field definition and object type definition should be built from DSL, not manually only dataFetcher should be supplied. We just need something to tell nadel that this fetcher should be used, how about:
`type Foo {
specialField: String <= $innerFetcher.specialFieldFetcher
}`
and then we could just do with only providing fetchers  -> no need to build types again etc
what do you think?